lib/src/conflicts.rs

```diff
@@ -259,15 +265,23 @@ impl ConflictMarkerKind {
     }
 }
 
+/// Represents a conflict marker parsed from the file. Conflict markers consist
+/// of a single ASCII character repeated for a certain length.
+struct ConflictMarker {
+    kind: ConflictMarkerKind,
+    len: usize,
+}
+
 /// Write a conflict marker to an output file.
 fn write_conflict_marker(
     output: &mut dyn Write,
     kind: ConflictMarkerKind,
+    len: usize,
     suffix_text: &str,
 ) -> io::Result<()> {
-    let conflict_marker: BString = iter::repeat(kind.to_byte())
-        .take(CONFLICT_MARKER_LEN)
-        .collect();
+    debug_assert!(len >= MIN_CONFLICT_MARKER_LEN);
```

Comment:
nit: I would make it `assert!()`, or remove the assertion. This expression is cheap. It's harmless to call this function with `len < MIN_CONFLICT_MARKER_LEN`.